src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -592,6 +592,14 @@ private function addService($id, $definition)
             $return[] = sprintf("@throws InactiveScopeException when the '%s' service is requested while the '%s' scope is not active", $id, $scope);
         }
 
+        if ($definition->isDeprecated()) {
+            if ($return && 0 === strpos($return[count($return) - 1], '@return')) {
+                $return[] = '';
+            }
+
+            $return[] = '@deprecated';
```

Comment:
For the end of the story, I changed my mind as I discussed with @nicolas-grekas, and we'd rather be stricter than necessary to be able to loose up the thing is needed.

Basically, I am now checking if the given template is valid enough (not containing `\r`, `\n`, `*/` symbols, and if provided, **must** contain `%service_id%` variable). In the XML and YAML dumpers, I also dump with `%service_id% as its "id" instead of `$id`, to be sure that the variable is there on the dump.

I'll update the docs saying that it is recomanded to change the deprecation message to be adapted on the specific use-case where the deprecation is activated.